lighthouse-core/audits/page-execution-timings.js

```diff
@@ -0,0 +1,161 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+/**
+ * @fileoverview Audit a page to see if it does not use <link> that block first paint.
+ */
+
+'use strict';
+
+const Audit = require('./audit');
+const Util = require('../report/v2/renderer/util');
+const DevtoolsTimelineModel = require('../lib/traces/devtools-timeline-model');
+
+const timelineCategories = [
+  {
+    name: 'Compile Script',
+    group: 'JavaScript',
+  },
+  {
+    name: 'Evaluate Script',
+    group: 'JavaScript',
+  },
+  {
+    name: 'Run Microtasks',
+    group: 'JavaScript',
+  },
+  {
+    name: 'Minor GC',
+    group: 'JavaScript',
+  },
+  {
+    name: 'Major GC',
+    group: 'JavaScript',
+  },
+  {
+    name: 'XHR Ready State Change',
+    group: 'JavaScript',
+  },
+  {
+    name: 'XHR Load',
+    group: 'JavaScript',
+  },
+  {
+    name: 'Layout',
+    group: 'Paint/Layout',
+  },
+  {
+    name: 'Paint',
+    group: 'Paint/Layout',
+  },
+  {
+    name: 'Composite Layers',
+    group: 'Paint/Layout',
+  },
+  {
+    name: 'Update Layer Tree',
+    group: 'Paint/Layout',
+  },
+  {
+    name: 'Recalculate Style',
+    group: 'Paint/Layout',
+  },
+  {
+    name: 'Parse HTML',
+    group: 'DOM/CSS',
+  },
+  {
+    name: 'Parse Stylesheet',
+    group: 'DOM/CSS',
+  },
+  {
+    name: 'DOM GC',
+    group: 'DOM/CSS',
+  },
+  {
+    name: 'Image Decode',
+    group: 'Images',
+  },
+];
+
+class PageExecutionTimings extends Audit {
+
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Performance',
+      name: 'page-execution-timings',
+      description: 'Page executiontime is high',
+      informative: true,
+      helpText: 'Consider reducing the time spent parsing, compiling and executing JS.' +
+        'You may find delivering smaller JS payloads helps with this.',
+      requiredArtifacts: ['traces'],
+    };
+  }
+
+  /**
+   * @param {!Array<TraceEvent>=} trace
+   * @param {!WebInspector.TimelineProfileTree.Node} A grouped and sorted tree
+   */
+  static getTimingsByCategory(trace) {
```

Comment:
this is pretty straightforward. you might as well inline this stuff into your `getExecutionTimingsByCategory` method

Reply:
will switch back to the old way, I tried override that methode to mock DevtoolsTimeLinemodel